### ISSUE TITLE
Configuration caching uses private java.base module APIs

### ISSUE THREAD

**Issue by ZacSweers** [gradle/gradle#19113]
### Expected Behavior
Gradle should either not use these or handle it accordingly

### Current Behavior

Configuration cache does not work in some cases on JDK 16+ due to module encapsulation

```
* What went wrong:
Configuration cache state could not be cached: field 'lineEndingsPolicy' from type 'com.diffplug.gradle.spotless.SpotlessTaskImpl': error writing value of type 'com.diffplug.spotless.extra.GitAttributesLineEndings$RelocatablePolicy'
> Configuration cache state could not be cached: field 'hasNonDefaultEnding' from type 'com.diffplug.spotless.extra.GitAttributesLineEndings$CachedEndings': error writing value of type 'com.googlecode.concurrenttrees.radix.ConcurrentRadixTree'
   > Configuration cache state could not be cached: field 'writeLock' from type 'com.googlecode.concurrenttrees.radix.ConcurrentRadixTree': error writing value of type 'java.util.concurrent.locks.ReentrantLock'
      > Unable to make field private final java.util.concurrent.locks.ReentrantLock$Sync java.util.concurrent.locks.ReentrantLock.sync accessible: module java.base does not "opens java.util.concurrent.locks" to unnamed module @6783dbc5

```

### Context

I reproduce this running the Spotless gradle plugin (6.0.0) with Gradle 7.3 and configuration caching enabled

### Steps to Reproduce 

See above

### Your Environment

Build scan URL:

**Comment by ZacSweers** [comment]
Let me know if this is actually a bug with Spotless (CC @nedtwigg)

**Comment by nedtwigg** [comment]
I just tried to reproduce on Java 17, but it worked for me. Does this always happen, or just sometimes? Does `rm -rf .gradle/configuration-cache` fix it?

Spotless supports configuration-cache by working around the serialization requirement, as described at https://github.com/diffplug/spotless/issues/987.

The field in question is [here](https://github.com/diffplug/spotless/blob/93c80ac5a670542177f6f7740e7107ba7536f483/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTask.java#L68-L77). It is not serializable, and uses the `LiveCache` workaround, which means that Spotless only works for a "live" daemon - the same daemon that executed the actual configuration phase.

We might need to make our workaround deeper, but I'm holding out hope that Gradle will re-examine the goals and design of the configuration cache feature.

**Comment by ZacSweers** [comment]
Yup still reproducible for me after. Fails without having a previous cache available (I.e. while creating a new cache)

**Comment by nedtwigg** [comment]
Happy to investigate further with a public repro.

**Comment by nedtwigg** [comment]
I believe this is fixed in the spotless grade plugin `6.1.0`.
